lib/plugins/paste.js

```diff
@@ -0,0 +1,29 @@
+import { Fragment, Slice } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+
+export default function paste(schema, options) {
+  return new Plugin({
+    props: {
+      transformPasted(slice) {
+        const child = slice.content?.firstChild?.firstChild;
+        if (!child || !child.isText) {
+          return slice;
+        }
+
+        const imageMatch = child.text.match(/\[Image: (.+)]/);
+        if (!imageMatch) {
+          return slice;
+        }
+
+        const image = schema.nodes.image.create({
+          markdown: child.text,
+        });
+        return new Slice(
+          Fragment.from(schema.nodes.paragraph.create({}, image)),
+          slice.openStart,
+          slice.openEnd,
+        );
+      },
+    },
+  });
+}
```

Comment:
Just to check my understanding - this plugin is needed because Prosemirror expects you to paste HTML in, not Govspeak, right?